### PR TITLE
feat: add `apm.setGlobalLabel()`

### PR DESCRIPTION
**Pull request by afharo**
In Kibana, we'd like to be able to update some of the global labels based on some dynamic information. The existing `apm.addLabels` or `apm.setLabel` only apply to the current transaction, but we need these specific labels to be applied to all the transactions.

Related to https://github.com/elastic/kibana/pull/157093

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update TypeScript typings
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Review by trentm — COMMENTED** [review]
Thanks!  This looks great and works for me. I used this example script to try it out: https://gist.github.com/trentm/bc7a6ae0cb849bccf3b87857963185b0

A couple suggested changes:

1. What do you think about *not* having this `return true`? I don't see the value. The existing `setLabel` only has that to handle there not being a current transaction. If so, then this patch would remove that:

```diff
diff --git a/index.d.ts b/index.d.ts
index e2ecc452..b3614ca4 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -116,7 +116,7 @@ declare namespace apm {
     currentSpan: Span | null;

     // Context
-    setGlobalLabel (name: string, value: LabelValue): boolean;
+    setGlobalLabel (name: string, value: LabelValue): void;
     setLabel (name: string, value: LabelValue, stringify?: boolean): boolean;
     addLabels (labels: Labels, stringify?: boolean): boolean;
     setUserContext (user: UserObject): void;
diff --git a/lib/agent.js b/lib/agent.js
index 1c378353..d11320be 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -368,7 +368,6 @@ Agent.prototype.setGlobalLabel = function (key, value) {
       return acc
     }, {})
   })
-  return true
 }

 Agent.prototype.setLabel = function (key, value, stringify) {
```


2. Here is an update to the agent.test.js that will have the agent send an actual transaction (and updated metadata) to a mock APM server. This allows the test to verify that the `this._transport.config(...)` call in the implementation did the necessary work.

patch: https://gist.github.com/trentm/c74e0e5ae5f626e67ad84116349eb6a3

my full test/agent.test.js if that is easier for you to apply: https://gist.github.com/trentm/5f571ecf1d55aa91d2b8b24823f99ff2

**Comment by afharo** [comment]
Thank you, @trentm! 🧡 

Your suggestions make perfect sense to me! I applied them in [`a8dab78` (#3337)](https://github.com/elastic/apm-agent-nodejs/pull/3337/commits/a8dab7865d38bc952dc35fd2010e8c4e53c4438a). Thank you for showing me _The way_!

**Comment by david-luna** [comment]
Thank you @afharo for your contribution. Just a small comment but not blocking

**Review by david-luna — APPROVED** [review]

File: test/agent.test.js [hunk 1]
```diff
@@ -1928,8 +2002,8 @@ test('patches', function (t) {
     agent.addPatch(moduleName, handler2)
     const modulePatches = agent._instrumentation._patches.get(moduleName)
     t.ok(modulePatches.length === 2 &&
-      modulePatches[0] === handler1 &&
-      modulePatches[1] === handler2, 'module patches are as expected')
+        modulePatches[0] === handler1 &&
+        modulePatches[1] === handler2, 'module patches are as expected')
```
[the remark this answers is not in the document]
Comment (afharo) [reply test/agent.test.js]: Oops! Not sure what happened here... probably my IDE behaving. I removed those extra spaces in my last commit 😇

**Pull request merged**